client/clientimpl_test.go

```diff
@@ -966,3 +968,253 @@ func TestRemoteConfigUpdate(t *testing.T) {
 		})
 	}
 }
+
+type packageTestCase struct {
+	name                string
+	errorOnCallback     bool
+	available           *protobufs.PackagesAvailable
+	expectedStatus      *protobufs.PackageStatuses
+	expectedFileContent map[string][]byte
+}
+
+const packageUpdateErrorMsg = "cannot update packages"
+
+func verifyUpdatePackages(t *testing.T, testCase packageTestCase) {
+	testClients(t, func(t *testing.T, client OpAMPClient) {
+
+		// Start a Server.
+		srv := internal.StartMockServer(t)
+		srv.EnableExpectMode()
+
+		localPackageState := internal.NewInMemPackagesStore()
+
+		var syncerDoneCh <-chan struct{}
+
+		// Prepare a callback that returns either success or failure.
+		onPackagesAvailable := func(ctx context.Context, packages *protobufs.PackagesAvailable, syncer types.PackagesSyncer) error {
+			if testCase.errorOnCallback {
```

Comment:
Why the error is returned when the flag is false rather than true?

Reply:
Oops. Fixed.